### PR TITLE
Only fallback to Discover if Following is first pinned

### PR DESCRIPTION
**Pull request by gaearon**
Removes the gate check since it isn't correct — the population in the split that are eligible for new onboarding haven't necessarily gone through the new onboarding.

Instead, use a heuristic. If the Following feed is the first pinned feed, keep the old behavior with the fallback. If it has been moved (manually or due to new onboarding), don't mix in Discover in there. This should roughly map to the old vs new onboarding, but doesn't depend on the gate.

## Test Plan

Add console logs to function creation. Verify putting Following first will create the "mixed in" instance, while putting something in front of it would create the "raw" following.

### CODE REVIEW

File: src/state/queries/post-feed.ts [hunk 1]
```diff
@@ -111,14 +110,18 @@ export function usePostFeedQuery(
   const enabled =
     opts?.enabled !== false && Boolean(moderationOpts) && Boolean(preferences)
   const userInterests = aggregateUserInterests(preferences)
+  const followingPinnedIndex =
+    preferences?.savedFeeds?.findIndex(
```
Comment (gaearon) [inline src/state/queries/post-feed.ts:114]: I assume these are safe to query (it'll convert old prefs on the fly?)

File: src/state/queries/post-feed.ts [hunk 2]
```diff
@@ -410,10 +409,10 @@ function createApi({
         userInterests,
       })
     } else {
-      if (useBaseFollowingFeed) {
-        return new FollowingFeedAPI({getAgent})
-      } else {
+      if (enableFollowingToDiscoverFallback) {
```
Comment (gaearon) [inline src/state/queries/post-feed.ts:412]: I found flipping this a little easier to think about, like an extra feature whose natural state is "off". Then `=== 0` condition above seems pretty plain, and doesn't need to be inverted.

**Review by haileyok — APPROVED** [review]
Tested on device, works correctly. Switching to following first gives me discover fallback while if I move it to last I get the "Pick some follows" copy.

**Review by pfrazee — APPROVED** [review]
Yep, tested, looks good

**Pull request merged**